lib/et_engine_connector.rb

```diff
@@ -20,8 +20,17 @@ def initialize(params = {})
 
   def stats(scenario_id)
     url = ETM_URLS[:stats] % [@provider, scenario_id]
+    params = @params
+
+    if params.key?(:keys) && params[:keys].is_a?(Array)
+      # 'stats' wants a hash of keys and attributes to fetch. If the caller does
+      # not care about specific attributes, request all.
+      #
+      # { keys: [:a, :b] } becomes { keys: { a: nil, b: nil } }
+      params = params.merge(keys: Hash[params[:keys].zip([])])
```

Comment:
This bit looks a bit hacky. At least - I have a hard time explaining it to my self. The comment above does explain it, but maybe it can become a bit more clear if `merge` isn't used. The `.zip([])` can be replaced by something else. Maybe do something like:

```ruby
params.store(:keys, params[:keys].each_with_object({}) do |key, obj|
  obj[key] = nil
end)
```

But I don't feel to strongly about this though.